module_5/oop_classes.py

```diff
@@ -0,0 +1,171 @@
+from datetime import date, datetime
+import random
+
+class CommonMethods:
+    """
+    Class contains common methods
+    """
+    def __create_output_file__(self, output_file):
+        """
+        Method to create output file
+        """
+        if not output_file:
+            open(output_file, 'a').close()
+
+    def populate_output_file(self, title, section_text, city_exp_date, calculated_val, output_file='/Users/shaliyu1/Documents/python_for_dqe/module_5/PDQE_newspaper.txt'):
+        self.__create_output_file__(output_file)
+        with open(output_file, "a") as myfile:
+            myfile.write(
+                f" {title} " + f"\n{section_text}" + f"\n{city_exp_date}" + f"\n{calculated_val}")
+
+    def get_current_date(self):
+        """
+        Method that returns current date
+        :return: current date
+        """
+        today = date.today()
+        return today
+
+
+class News:
+    """
+    Class contains methods related to News section
+    """
+
+    def get_news(self):
+        """
+        Method to get news and city from user
+        :return: news and city
+        """
+        print("\nNews will be added to next edition")
+        news = input("""\nPlease enter your news: """)
+        news_city = input("""\nPlease enter your city: """)
+        return news, news_city
+
+class Ads(CommonMethods):
+    """
+    Class contains methods related to advertisement section
+    """
+
+    common_methods = CommonMethods()
+
+    def get_ads(self):
+        """
+        Method to get input regarding ads fron user
+        :return: ads text, expiration_date
+        """
+        print("\nAdvertising will be added to next edition")
+        ads = input("""\nPlease enter your ads: """)
+        expiration_date = input("""\nPlease enter expiration date in format 'yyyy-mm-dd': """)
```

Comment:
If you enter the date in the wrong format, the code will not work.
Process this variant with try.

Reply:
done